third_party/cached_httpfs/include/http_file_cache.hpp

```diff
@@ -78,6 +80,13 @@ class CachedFileHandle {
 	shared_ptr<CachedFile> file;
 };
 
+class LocalCacheFileSystem: public LocalFileSystem {
+	// TODO: we could lock down the LocalFileSystem to only allow path that are in the cache directory
```

Comment:
Would be nice to implement this indeed as a bit of defense in depth. Seems like it should only be necessary to override `OpenFile`, with one that first resolves symlinks and `../..` and then checks the prefix.

Reply:
Yeah definitely.